projects/ancestor/ancestor.py

```diff
@@ -0,0 +1,79 @@
+class Stack():
+    def __init__(self):
+        self.stack = []
+    def push(self, value):
+        self.stack.append(value)
+    def pop(self):
+        if self.size() > 0:
+            return self.stack.pop()
+        else:
+            return None
+    def size(self):
+        return len(self.stack)
+
+class Graph():
+    def __init__(self):
+        self.vertices = {}
+
+# --------------------------------------------------------------------
+
+    def add_vertex(self, vertex):
+        if vertex not in self.vertices:
+            self.vertices[vertex] = set()
+
+# --------------------------------------------------------------------
+
+    def add_edge(self, v1, v2):
+        if v1 in self.vertices and v2 in self.vertices:
+            self.vertices[v2].add(v1)   
+        else:
+            raise IndexError("That vertex doesn't exist")
+
+# --------------------------------------------------------------------
+
+    def earliest_ancestor(self, start):
+        stack = Stack()
+        stack.push([start])
+        visited = set()
+        new_path = None
+        while stack.size() > 0:
+            path = stack.pop()
+            node = path[-1]
+            if node not in visited:
+                visited.add(node)
+                for next_node in self.vertices[node]:
+                    new_path = path.copy()
+                    new_path.append(next_node)
+                    stack.push(new_path)
+        if new_path == None:
+            return -1
+        else:
+            return new_path[-1]
+
```

Comment:
Nice work with this ancestors problem.  Since the test file was added late, it is understandable that this is implemented in a way that does not match with it.  But it seems to be working well from what i can tell!.